lib/court_data_adaptor/resource/base.rb

```diff
@@ -3,8 +3,36 @@
 module CourtDataAdaptor
   module Resource
     class Base < JsonApiClient::Resource
+      include Configurable
+
       VERSION = '0.0.1'
-      self.site = ENV['COURT_DATA_ADAPTOR_API_URL']
+      self.site = config.api_url
+
+      def self.bearer_token
+        config.test_mode? ? fake_bearer_token : client_bearer_token
+      end
+
+      def self.client_bearer_token
+        Client.new.bearer_token
+      end
+
+      def self.fake_bearer_token
+        'fake-court-data-adaptor-bearer-token'
+      end
+
+      connection do |conn|
+        conn.use(
+          FaradayMiddleware::OAuth2,
+          bearer_token,
+          token_type: :bearer
+        )
+
+        # example setting response logging
```

Comment:
If PR no longer a w.i.p, please remove.

Reply:
these comments were in the previous connection blocks location. I kept them as a reminder to consider logging and other options at a later date.